src/main/java/net/dv8tion/jda/api/requests/restaction/pagination/PaginationAction.java

```diff
@@ -261,6 +263,112 @@
+     *
+     * @see    #takeWhileAsync(int, Predicate)
+     * @see    #takeUntilAsync(Predicate)
+     */
+    default CompletableFuture<List<T>> takeWhileAsync(final Predicate<? super T> rule)
+    {
+        Checks.notNull(rule, "Rule");
+        return takeUntilAsync(rule.negate());
+    }
+
+    /**
+     * Retrieves elements while the specified condition is met.
+     *
+     * @param  limit
+     *         The maximum amount of elements to collect or {@code 0} for no limit
+     * @param  rule
+     *         The rule which must be fulfilled for an element to be added,
+     *         returns false to discard the element and finish the task
+     *
+     * @throws IllegalArgumentException
+     *         If the provided rule is {@code null} or the limit is negative
+     *
+     * @return {@link CompletableFuture} - Type: {@link List List}
+     *         <br>Future representing the fetch task, the list will be sorted most recent to oldest
+     *
+     * @see    #takeWhileAsync(Predicate)
+     * @see    #takeUntilAsync(int, Predicate)
+     */
+    default CompletableFuture<List<T>> takeWhileAsync(int limit, final Predicate<? super T> rule)
+    {
+        Checks.notNull(rule, "Rule");
+        return takeUntilAsync(limit, rule.negate());
+    }
+
+    /**
+     * Retrieves elements until the specified condition is met.
+     *
+     * @param  rule
+     *         The rule which must be fulfilled for an element to be discarded,
+     *         returns true to discard the element and finish the task
+     *
+     * @throws IllegalArgumentException
+     *         If the provided rule is {@code null}
+     *
+     * @return {@link CompletableFuture} - Type: {@link List List}
+     *         <br>Future representing the fetch task, the list will be sorted most recent to oldest
+     *
+     * @see    #takeWhileAsync(Predicate)
+     * @see    #takeUntilAsync(int, Predicate)
+     */
+    default CompletableFuture<List<T>> takeUntilAsync(final Predicate<? super T> rule)
+    {
+        return takeUntilAsync(0, rule);
+    }
+
+    /**
+     * Retrieves elements until the specified condition is met.
+     *
+     * @param  limit
+     *         The maximum amount of elements to collect or {@code 0} for no limit
+     * @param  rule
+     *         The rule which must be fulfilled for an element to be discarded,
+     *         returns true to discard the element and finish the task
+     *
+     * @throws IllegalArgumentException
+     *         If the provided rule is {@code null} or the limit is negative
+     *
+     * @return {@link CompletableFuture} - Type: {@link List List}
+     *         <br>Future representing the fetch task, the list will be sorted most recent to oldest
+     *
+     * @see    #takeWhileAsync(Predicate)
+     * @see    #takeUntilAsync(int, Predicate)
+     */
+    default CompletableFuture<List<T>> takeUntilAsync(int limit, final Predicate<? super T> rule)
```

Comment:
The `limit` should come after the `rule` as there is a method that only takes the rule.

Reply:
This is for better kotlin support. Having the limit be the second parameter would make it look like this in kotlin `action.takeUntilAsync({ condition(it) }, 10)` compared to now `action.takeUntilAsync(10) { condition(it) }`.

```kotlin
action.takeUntilAsync(1000) {
    it.idLong > startId
}
```